### PR TITLE
Add roadmap to kueue website

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind documentation

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by alculquicondor** [comment]
I prefer leaving this tentative roadmap in the README.

**Comment by alculquicondor** [comment]
@ahg-g, thoughts?

**Comment by ahg-g** [comment]
No strong thoughts

**Comment by moficodes** [comment]
What other projects have their roadmap on their docsite?

I am not sure if thats the best place for it.

github repo is probably better.

**Comment by kerthcet** [comment]
Adding a roadmap to Website is not a rare matter, like https://kubevela.io/docs/roadmap/. But respect all your advices, feel free to close this if you think not necessary right now. ^_^

**Comment by alculquicondor** [comment]
let's come back to this after the 0.3 release.

File: site/content/en/docs/roadmap/_index.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - Ray
+  - Workflows (Tekton, Argo, etc.)
+
+### Roadmap for long-term
+
+These features are in no particular order:
+
+- Budget support [#28](https://github.com/kubernetes-sigs/kueue/issues/28)
+- Dashboard for management and monitoring for administrators
+- Multi-cluster support
```
Comment (ArangoGutierrez) [inline site/content/en/docs/roadmap/_index.md:30]: nit: new line at the end of file

**Review by ArangoGutierrez — COMMENTED** [review]
overall I agree, README should be for a quick start of the project, a short intro and that's it, for everything else README should offer links to documentation and other topics of interest

**Comment by kerthcet** [comment]
At the least, the roadmap should be updated I think.

File: site/content/en/docs/roadmap/_index.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    This page describes a high-level overview of the main priorities for the project.
+---
+
+### Roadmap for 2023
+
+These features are in expected order of release:
+
+- Job preemption to reclaim borrowed quota and to accommodate high priority jobs [#83](https://github.com/kubernetes-sigs/kueue/issues/83), this is planned for v0.3
+- Cooperative preemption support for workloads that implement checkpointing [#477](https://github.com/kubernetes-sigs/kueue/issues/477)
+- Flavor assignment strategies, e.g. _minimizing cost_ vs _minimizing borrowing_ [#312](https://github.com/kubernetes-sigs/kueue/issues/312)
```
Comment (alculquicondor) [inline site/content/en/docs/roadmap/_index.md:16]: I don't think these two are being prioritized at all

**Pull request closed without merging**